src/pages/auth/login.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  GetServerSideProps,
+  InferGetServerSidePropsType,
+  NextPage,
+} from "next";
+import { getCsrfToken } from "next-auth/react";
+import Link from "next/link";
+import { useRouter } from "next/router";
+
+import { getServerAuthSession } from "@server/common/getServerAuthSession";
+
+const Login: NextPage = ({
+  csrfToken,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) => {
+  const router = useRouter();
+  return (
+    <div>
+      <form method="post" action="/api/auth/callback/credentials">
+        <input name="csrfToken" type="hidden" defaultValue={csrfToken} />
+        <label>
+          Email
+          <input name="email" type="email" />
+        </label>
+        <label>
+          Password
+          <input name="password" type="password" />
+        </label>
+        <button type="submit">Sign in</button>
+      </form>
+      <Link
+        href={{
+          pathname: "/auth/register",
+          query: { callbackUrl: router.query.callbackUrl },
+        }}
+      >
+        <button>Create Account</button>
+      </Link>
+    </div>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const callbackUrl = context.query.callbackUrl;
```

Comment:
Can this be undefined/null? `context?.query?.callbackUrl`

Reply:
The`.callbackUrl` part can be undefined/null but that is fine because the place that uses it checks `destination: callbackUrl ?? "/",`. `context.query` will always be present (reference: https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props#context-parameter).  Also typescript would type error here if it was not guaranteed.